### ISSUE TITLE
Add  predict_proba method to differentiable head

### ISSUE THREAD

**Issue by OskarLiew** [huggingface/setfit#168]
Right now there does not seem to be any easy way to get prediction probabilities from `SetFitModel` when using the differentiable head.

Example code:

```python
from setfit import SetFitModel

# Load a SetFit model from Hub
model: SetFitModel = SetFitModel.from_pretrained(
    "sentence-transformers/paraphrase-mpnet-base-v2",
    use_differentiable_head=True,
    head_params={"out_features": 2},
)

preds = model.predict_proba("Pineapple on pizza is delicious 🍕🍍😍")  # Try it!
print(preds)
```

Running the above code results in the following error

```
AttributeError: 'SetFitHead' object has no attribute 'predict_proba'. Did you mean: 'predict_prob'?
```

**Comment by nbertagnolli** [comment]
This looks like a typo in the head I made a PR here which should fix it: #170. @blakechi could you take a look : )

**Comment by blakechi** [comment]
Just checked. It looks nice, thank you both!

**Comment by OskarLiew** [comment]
Closed in #171
